Bottom: 2eee8982fcd1e14263d58ade8080438507ed1862
Top:    ae9f4dbeacfeb5bd98658f9a56935c7e8a41566b
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2013-01-03 17:10:47 +0000

find.c: Fix possible un-assigned use of "flag" in DRCFind()

Caught by inspection.. no idea why the compiler didn't notice.

Before commit 4cbfadfb7a496e60c25f47c88bdb374b4d3777bb, our local "flag"
variable was the file-global "TheFlag". If the "shrink" test was not run,
the code called ClearFlagOnAllObjects() with the previously assigned
"TheFlag". After the above commit, we would just use "flag" unassigned.

In this case, FOUNDFLAG | SELECTEDFLAG are probably the correct flags to
clear, as it matches the final cleanup after the second test in DRCFind().

This changes the behaviour of the ClearFlagOnAllObjects(), as up to this
commiti, when the "shrink" test is run, the last "flag" used was FOUNDFLAG.


---

diff --git a/src/find.c b/src/find.c
index e63572e..9198df4 100644
--- a/src/find.c
+++ b/src/find.c
@@ -3396,7 +3396,13 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
     }
   /* now check the bloated condition */
   drc = false;
+  flag = FOUNDFLAG | SELECTEDFLAG;
   ClearFlagOnAllObjects (false, flag);
+  Bloat = 0;
+  flag = SELECTEDFLAG;
+  ListStart (What, ptr1, ptr2, ptr3, flag);
+  DoIt (flag, true, false);
+  DumpList ();
   flag = FOUNDFLAG;
   ListStart (What, ptr1, ptr2, ptr3, flag);
   Bloat = PCB->Bloat;
